src/pages/index.js

```diff
@@ -23,70 +31,121 @@ const addNewCardButton = document.querySelector(".profile__add-button");
+/*                                                                     Functions                                                                                                                                  */
 /*--------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+api
+  .getCards()
+  .then((initialCards) => {
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => console.error(err));
 
-const previewImageModal = document.querySelector("#modal-preview");
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
 
-const previewModalCloseButton =
-  previewImageModal.querySelector(".modal__close");
+api
+  .getUsers()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    userInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) => console.error(err));
 
-/*--------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
-/*                                                                                                                                                                          */
-/*                                                                     Event Handlers                                                                                                                                                 */
-/*                                                                                                                                                                          */
-/*--------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
 function handleProfileEditSubmit(formValues) {
   const nameInput = formValues.title;
   const descriptionInput = formValues.description;
-  userInfo.setUserInfo({
-    name: nameInput,
-    description: descriptionInput,
-  });
-  profileModal.close();
-}
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
-});
+  profileModal.setLoading(true);
+
+  api
+    .editUserInfo(nameInput, descriptionInput)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: nameInput,
+        description: descriptionInput,
+      });
+      profileModal.close();
+    })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**